source/StringUtility.cpp

```diff
@@ -279,12 +279,12 @@ namespace oldstrutil
 		}
 		throw std::runtime_error( oldstrutil::format( "Section index %ul exceeded size %ul from sections %s", sectionindex, sec.size(), value.c_str() ));
 	}
-	
+
 	//+++++++++++++++++++++++++++++++++++++++++
 	std::string extractSection( const std::string& value, const std::string& sep, std::string::size_type start, std::string::size_type stop )
 	{
 		std::string data;
-		int count = -1;
+		size_t count = 0;
```

Comment:
We should find another way rather than making `start`/`stop` signed considering:
1. `std::string::size_type` is a type that guarantees `start`/`stop` will be large enough to contain the number of characters within a string. This type is unsigned.
2. The default value of `stop` in the `extractSection()` declaration is set to `std::string::npos` which is also unsigned.

What about incrementing at the end of the loop instead? That way, the first pass is zero as expected. Another option could be to perform a check and only increment if `start` is greater than 0 (zero).

Reply:
@Xoduz && @strattond: Did you see my question and comment 2 days ago regarding 2 other possible workarounds?
```
What about incrementing at the end of the loop instead? That way, the first pass is zero as expected. Another option could be to perform a check and only increment if start is greater than 0 (zero).
```

Reply:
An end of loop increment will probably work yes.  Better than conditional logic up front.

Reply:
Fixed in 4ce27d8.